Capstone/Assets/Game/Scripts/StateMachine/EnemyStates/EnemyFSMState_SEEK.cs

```diff
@@ -4,35 +4,101 @@
 
 namespace Bladesmiths.Capstone
 {
+    /// <summary>
+    /// The Seeking behavior for the Enemy
+    /// </summary>
     public class EnemyFSMState_SEEK : EnemyFSMState
     {
         private Player _player;
         private Enemy _enemy;
+        private float sideMoveTimer;
+        private float sideMoveTimerMax;
+        private float seekAgainTimer;
+        private float seekAgainTimerMax;
+        private CharacterController controller;
+        private float speed;
+        private float dir;
 
         public EnemyFSMState_SEEK(Player player, Enemy enemy)
         {
             _player = player;
             _enemy = enemy;
-
         }
 
         public override void Tick()
         {
-            Vector3 dist = _player.transform.position + Vector3.up - _enemy.transform.position;
-
-            _enemy.gameObject.transform.position += dist.normalized * Time.deltaTime;
-
+            MovementCheck();
         }
 
         public override void OnEnter()
         {
+            sideMoveTimer = 0;
+            sideMoveTimerMax = 1f;
+            seekAgainTimerMax = 1f;
+            seekAgainTimer = seekAgainTimerMax;
+            controller = _enemy.GetComponent<CharacterController>();
+            speed = 1.5f;
+            dir = Random.Range(-1, 2);
 
         }
 
         public override void OnExit()
         {
 
         }
+        
+        /// <summary>
+        /// Runs the movement code for seeking the player
+        /// </summary>
+        public void MovementCheck()
+        {
+            Vector3 movementVector = Vector3.zero;
+
+            Vector3 dist = _player.transform.position - _enemy.transform.position;
+
+            // If the Enemy is within 2 units and the seekAgainTimer isn't started
+            if (dist.magnitude > 2f && seekAgainTimer == seekAgainTimerMax)
```

Comment:
I feel that the enemies get too close to the player when surrounding. Try increasing this a little bit to give the player a bit of space